client/components/deposits-overview/recent-deposits-list.tsx

```diff
@@ -24,26 +24,26 @@ import { CachedDeposit } from 'wcpay/types/deposits';
 import { formatCurrency } from 'wcpay/utils/currency';
 import { getDetailsURL } from 'wcpay/components/details-link';
 
-interface DepositRowProps {
-	deposit: CachedDeposit;
-}
-
 interface RecentDepositsProps {
 	deposits: CachedDeposit[];
 }
 
 const tableClass = 'wcpay-deposits-overview__table';
 
 /**
- * Renders a recent deposits table row.
+ * Renders the Recent Deposit list component.
  *
- * @return {JSX.Element} Deposit table row.
+ * This component includes the recent deposit heading, table and notice.
  */
-const DepositTableRow: React.FC< DepositRowProps > = ( {
-	deposit,
-} ): JSX.Element => {
-	return (
-		<Flex className={ `${ tableClass }__row` }>
+const RecentDepositsList: React.FC< RecentDepositsProps > = ( {
+	deposits,
+} ) => {
+	if ( deposits.length === 0 ) {
+		return null;
+	}
+
+	const depositRows = deposits.map( ( deposit ) => (
+		<Flex className={ `${ tableClass }__row` } key={ deposit.id }>
```

Comment:
Just noticed that `tableClass` is a global. Suggest moving the variable inside this component, or possibly hard-code all the classNames directly. Also curious about best practice tips here. I have a vague preference for using full class so easier to find relevant css, or only constructing `className`s when absolutely necessary e.g. when it's dynamic based on user input or maybe on parent / client component needs.

Reply:
> Just noticed that `tableClass` is a global. 

I'm curious about your thoughts on the downsides of this. I understand `tableClass` to be local to this file, not global to any other execution context.

This also aligns with how I see the entire file as a self-contained component. No need to define vars like this inside the function that may be re-declared on each render – I probably also have an underlying fear of this due to prior traumatic experiences with `useEffect`.

> I have a vague preference for using full class so easier to find relevant CSS

💯 Me too, this is additional complexity and harder to read – we can find/replace it when a change is needed.

Taking another look, I'm unsure if we use this for styling at all – so we might be able to remove the `className`s in this component entirely.

Reply:
One reason that `${ tableClass }__row` may be preferable is that it matches the scss-style `&__row`, [for this component at least](https://github.com/Automattic/woocommerce-payments/blob/c5e4b72d2ebf2b09c92171908bd0c0ca702b4982/client/components/deposits-overview/style.scss/#L55-L64).

```scss
&__table {
		&__row {
			&__header {
				font-weight: 600;

				div {
					padding: 8px 0;
				}
			}
		}
```

IMO this is harder to parse/search at a glance than `wcpay-deposits-overview__table__row`, which is why I made the mistake earlier of thinking these classes weren't used for much 😆

Reply:
> I'm curious about your thoughts on the downsides of this. I understand tableClass to be local to this file, not global to any other execution context.

File-local is bigger scope than function-local, so I'd go with the [smallest scope necessary](https://google.github.io/styleguide/cppguide.html#Local_Variables). Reduce coupling, more self-contained.

This is all pretty low priority but also good to align on as a code style guide.

The `&__row` rationale for using a variable makes sense to me 👍🏻

Reply:
> File-local is bigger scope than function-local, so I'd go with the [smallest scope necessary](https://google.github.io/styleguide/cppguide.html#Local_Variables). Reduce coupling, more self-contained.

I agree with this, but... the ~magic~weirdness of React components makes this a tradeoff. I'd be curious to see what Google say about this in their React style guide.

The function declared inside the component will be re-defined each time the component renders. This is usually no big deal but becomes a mess when you want to pass this function into a hook or a child component and the next minute, you've triggered an infinite re-render or something.

You'll probably end up fixing this by wrapping it in `useCallback` _if it needs the local component state_.

If, however, you have a **pure function** that doesn't rely on the component's state, there is no need to worry about any of this – define the function outside the component and you're good to go. Less to think about.

Reply:
> The function declared inside the component will be re-defined each time the component renders. This is usually no big deal but becomes a mess when you want to pass this function into a hook or a child component and the next minute, you've triggered an infinite re-render or something.

Hmm … I hadn't considered that. Are you referring to this anon function here:

```jsx
deposits.map( ( deposit ) => ( /* this function */ ) )
```

Since that function is not going to be pulled out (if it was, I'd trust future dev to deal with any risks – would the React lint tools spot this?), I would still prefer declaration closer to use, e.g. 

```jsx
const tableClass = 'wcpay-deposits-overview__table';
	
const depositRows = deposits.map( ( deposit ) => (
  <Flex className={ `${ tableClass }__row` } key={ deposit.id }>
```

…but I trust your experience here, your call :)

Reply:
> Hmm … I hadn't considered that. Are you referring to this anon function here:

😂 I've gone down a rabbit-hole and am not even talking about this component anymore. Your suggestion is not a problem at all, but I understand why the author chose to define this var outside. In this case, let's bring it inside!

Some folks prefer to define outside unless it requires component state (I lean towards this, but mainly for functions) – some prefer to keep things defined inside, closer to where it is used unless it would cause a problem (I think this is what you're leaning to).

Reply:
> if it was, I'd trust future dev to deal with any risks – would the React lint tools spot this?)

I tested this and yes, you are right, I get a big, obvious linter error:

```
The ... function makes the dependencies of useEffect Hook (at line 67) change on every render.
To fix this, wrap the definition of ... in its own useCallback() Hook.
eslint [react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
```

I can sleep at night knowing the linter has my back (or can I?)